File: java-workshop/part-4-behavior/src/main/java/service/OrderResult.java
package service;

import domain.Order;
import domain.ShippingConfirmation;

public sealed interface OrderResult permits OrderResult.Success, OrderResult.Failure {
    record Success(Order order, ShippingConfirmation shippingConfirmation) implements OrderResult {
    }

    record Failure(Order order, String reason) implements OrderResult {
    }
}

File: csharp-workshop/Part3.MocksClean/Service/OrderResult.cs
using Part3.MocksClean.Domain;

namespace Part3.MocksClean.Service;

public abstract record OrderResult
{
    public sealed record Success(Order Order, ShippingConfirmation ShippingConfirmation) : OrderResult;
    public sealed record Failure(Order Order, string Reason) : OrderResult;
}
